feat: Configure AI interview chat with career conversation questions

granit3 opened the pull request:
Implements AI-powered interview chat that uses curated career questions as conversational guidance. The AI maintains a natural, flowing dialogue while exploring important themes about user goals, values, direction, and potential fit with opportunities.

Backend (Python API):
- Add /api/interview/chat endpoint using Gemini 2.5 Flash
- Load career_conversation_questions.json with 8 goal categories
- Build system prompt that guides AI to use questions naturally (not verbatim)
- Handle conversation history for context-aware responses
- Add Pydantic models for request/response validation

Frontend:
- Integrate Next.js API route with Python API
- Add Docker environment detection for correct API URL selection
- Send conversation history to backend for context
- Update initial AI message to be more welcoming
- Add suppressHydrationWarning to prevent Next.js warnings

Security & Configuration:
- Move GEMINI_API_KEY to .env file
- Add env_file configuration in docker-compose.yml
- Add PYAPI_URL environment variable for lv-web service

granit3 committed:
feat: Configure AI interview chat with career conversation questions

Implements AI-powered interview chat that uses curated career questions
as conversational guidance. The AI maintains a natural, flowing dialogue
while exploring important themes about user goals, values, direction, and
potential fit with opportunities.

Backend (Python API):
- Add /api/interview/chat endpoint using Gemini 2.5 Flash
- Load career_conversation_questions.json with 8 goal categories
- Build system prompt that guides AI to use questions naturally (not verbatim)
- Handle conversation history for context-aware responses
- Add Pydantic models for request/response validation

Frontend:
- Integrate Next.js API route with Python API
- Add Docker environment detection for correct API URL selection
- Send conversation history to backend for context
- Update initial AI message to be more welcoming
- Add suppressHydrationWarning to prevent Next.js warnings

Security & Configuration:
- Move GEMINI_API_KEY to .env file
- Add env_file configuration in docker-compose.yml
- Add PYAPI_URL environment variable for lv-web service

Code Quality:
- Remove debug console.logs
- Remove commented-out code
- Remove unused timestamp from conversation history

hilalnurgercek reviewed (commented):
Reviewed. I see that the unit tests are missing from this pull request. I think they should also be here for it to be considered "Done" according to the DoD.

Aaryan-nyr commented:
> Reviewed. I see that the unit tests are missing from this pull request. I think they should also be here for it to be considered "Done" according to the DoD.

Hi, just out of curiosity how have you reviewed this?

hilalnurgercek commented:
> > Reviewed. I see that the unit tests are missing from this pull request. I think they should also be here for it to be considered "Done" according to the DoD.
> 
> Hi, just out of curiosity how have you reviewed this?

Just checked if it is according to the team conventions and if the unit tests are there.

hilalnurgercek commented:
I will test on the feature branch and update the review.

Aaryan-nyr commented:
> I will test on the feature branch and update the review.

thanks

Aaryan-nyr commented on apps/lv-web/src/app/api/interview/chat/route.ts:
```diff
@@ -15,29 +14,66 @@ export async function POST(request: NextRequest) {
+    // This is more reliable than checking for container existence
+    const isDocker = process.env.DATABASE_URL?.includes('@db:');
+    const apiUrl = isDocker 
+      ? 'http://lv-pyapi:8080'  // Docker internal network
```
Hi, what is the reason for this expression? 
is it if the env variable PYAPI_URL is not set / corrupted somehow? 
Moreover, can you assign the URL to a variable and then use those? 
PYAPI_URL=http://...

Aaryan-nyr commented on apps/lv-pyapi/main.py:
```diff
@@ -62,16 +80,75 @@ async def get_user(user_id: str, db: Session = Depends(get_db)):
         return JSONResponse(status_code=500, content={"message": str(e), "status": 500})
 
+@app.post("/api/interview/chat")
+async def interview_chat(request: InterviewChatRequest) -> dict[str, Any]:
```
where is the function called?

Aaryan-nyr commented:
I see that an entirely new function was added. Are there any tests for it? Could they be with the Gemini API tests?

Aaryan-nyr commented:
More general question: How can I run this locally?

granit3 committed:
test: AI interview chat

hilalnurgercek commented:
> More general question: How can I run this locally?

i would also need help on this one

granit3 commented:
Driving back to Vantaa rn. Isnt it working via Docker?

CatSpectre commented:
I havent reviewed locally yet but I think pyapi should have .env.example so its more clear

granit3 committed:
refactor: Simplify API URL config and update documentation

- Simplify route.ts to use PYAPI_URL directly (removed Docker detection logic)
- Update documentation with Gemini API key setup instructions
- Clean up code style in main.py (removed unnecessary type hints)
- Update prompt building to use json.dumps for career questions (now shorter)
- Remove suppressHydrationWarning from layout
- Add .env.local cleanup in docker-compose for lv-web
- Add test run command comment in test file

johottaja commented on apps/lv-web/src/app/api/interview/chat/route.ts:
```diff
@@ -15,29 +14,57 @@ export async function POST(request: NextRequest) {
 
     const body = await request.json();
-    const { message } = body;
+    const { message, conversation_history } = body;
```
I don't think we should be passing the conversation history back and forth in the chat HTTPS requests.

johottaja replied:
Seems inefficient but the chats wouldn't be that long so I guess it's okay if we don't want to do it properly. Other than this, the commit looked fine, main.py was maybe a bit cluttered with the class definitions and the prompt but nothing too serious.

LKdev03 commented on apps/lv-web/README.md:
````diff
@@ -50,6 +50,12 @@ A simple, clean fullstack template built with Next.js, NextAuth.js, Prisma, and
    GITHUB_CLIENT_ID="your-github-client-id"
    GITHUB_CLIENT_SECRET="your-github-client-secret"
    ```
+   # Set up a new .env file in the lv-pyapi.
+   # Set up your Gemini API key there:
+   # GEMINI_API_KEY=YOUR_GEMINI_API_KEY_HERE
+   # And the key you'll get from https://aistudio.google.com/app/api-keys 
````
Just a readme file but you can't find the API used in this project from google AI studio if you don't import the key from gcp there. 

So in other words since the key is already created on GCP this link should be changed to:
https://console.cloud.google.com/apis/credentials?project=swp-livingvectors
